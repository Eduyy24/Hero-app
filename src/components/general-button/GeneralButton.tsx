import { FunctionComponent } from 'react';
import styles from './GeneralButton.module.css'

type Props = {
  name: string;
  onClick: EmptyFunction;
}
export const GeneralButton: FunctionComponent<Props> = (props) => {
  const {name = '', onClick} = props;
  return <button onClick={onClick} name={name} className={styles.generalButton}>{name}</button>
}