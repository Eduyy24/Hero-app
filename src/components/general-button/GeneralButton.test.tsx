import {render, fireEvent, screen} from '@testing-library/react';
import {GeneralButton} from './GeneralButton'
describe('<GeneralButton />', () => {
  test('onClick called', () => {
    const name = 'Aceptar'
    const onClick = jest.fn()

    render(<GeneralButton name={name} onClick={onClick} />)
    const btn = screen.getByText(name)

    fireEvent.click(btn)
    expect(onClick).toHaveBeenCalled()
  })
})