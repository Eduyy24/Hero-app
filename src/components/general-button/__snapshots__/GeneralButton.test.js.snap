// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeneralButton /> onClick called 1`] = `
<button
  class="generalButton"
  name="Aceptar"
>
  Aceptar
</button>
`;
